pages/intro.js

```diff
@@ -0,0 +1,186 @@
+import Card, { CardHeader, CardTitle, CardContent, CardButtonContainer } from '../components/card'
+import styled from 'styled-components'
+import { useState } from 'react'
+import { COLOR, EDIT } from '../constants'
+import Button from '../components/button'
+import TextBox from '../components/textbox'
+import {db} from '../utility/firebase'
+
+const HeaderText = styled.p`
+    border: none;
+    overflow: hidden;
+    background-color: ${COLOR.BACKGROUND};
+    outline: none;
+    font-size: 24px;
+    padding-bottom: 13px;
+    color: ${COLOR.TEXT};
+    font-weight: 600;
+`
+
+const ContentText = styled.p`
+    border: none;
+    overflow: hidden;
+    background-color: ${COLOR.BACKGROUND};
+    outline: none;
+    color: ${COLOR.TEXT};
+`
+
+const StyledCancel = styled.button`
+    font-size: 16px;
+    cursor: pointer;
+    border-bottom: 2px solid ${COLOR.BLACK};
+    margin-left: 675px;
+    margin-right: 40px;
+    border: none;
+    outline: none;
+    background-color: ${COLOR.BACKGROUND};
+`
+
+const StyledHeader = styled.p`
+    padding-bottom: 13px;
+    padding-top: 17px;
+    margin: 0px;
+    background-color: ${COLOR.BACKGROUND};
+`
+
+export default function IntroPage() {
+    const [isEditingObj, setIsEditingObj] = useState({})
+    const [websiteData, setWebsiteData] = useState({})
+    // TODO need functionality to display data based on what event is currently being viewed
+
+    // TODO need to substitute the value at .doc(ID) with current website
+    const getHackathonData = async () => {
+        await db.collection('Hackathons').doc('IntroTestHackathon').get().then((doc) => {
+            setWebsiteData(doc.data().WebsiteData)
+        })
+    }
+
+    if (Object.keys(websiteData).length == 0) {
+        getHackathonData()
+    }
+
+    const handleEdit = (type) => {
+        setIsEditingObj({
+            ...isEditingObj,
+            [type]: !isEditingObj[type]
+        })
+        
+    };
+
+    const handleCancel = (type) => {
+        setIsEditingObj({
+            ...isEditingObj,
+            [type]: false
+        })
+    }
+
+    /**
+     * @param {string} type every section in the intro has a type, which is the key in the map of WebsiteData
+     * reads the value of the header/content associated with the given type and updates the object in Firebase
+     */
+    const handleSave = async (type) => {
+        const textareas = document.getElementsByClassName(type)
+        const time = new Date(new Date().getTime())
+        let updateObj = {
+            WebsiteData: {
+                ...websiteData,
+                [type]: {
+                    editor: 'GRAB USER FROM REDUX?',
```

Comment:
maybe make a variable at the top with this string, so that we can easily change it later.